apollo-router-core/src/spec/query.rs

```diff
@@ -23,15 +26,25 @@ impl Query {
     ///
     /// This will discard unrequested fields and re-order the output to match the order of the
     /// query.
-    #[tracing::instrument(level = "trace")]
-    pub fn format_response(&self, response: &mut Response, operation_name: Option<&str>) {
+    #[tracing::instrument(skip_all, level = "trace")]
+    pub fn format_response(
```

Comment:
Maybe followup: Is there a chance this can be done progressively in the execute steps?

Reply:
Oh by progressively I meant chunks by chunks as they arrive, but I might be missing context here

Reply:
It's kinda the atomic operation here already as the `ResponseStream` yields `Reponse`s and this function applies on a single `Response`.

Since there is no IO or wait on network or something, this function has nothing to "await" for, so it needs to be blocking. Though a caller could probably use tokio::spawn_blocking if they do notice the processing is taking too long.

There might be also a case to make about limiting the size of this so it doesn't block the entire application indefinitely when the payload is unreasonably big. Though I suspect we might want to do that on the request first or everywhere there is JSON parsing... I'm not sure how serde handles limit in payload. Maybe we should limit the query size in bytes? And response size from the subgraphs?